File: python/04.py
import time

# Create a large string as sample data
data = "Lorem ipsum dolor sit amet, consectetur adipiscing elit."

# Start the timer
start = time.time()

# Perform string processing operations here
reversed_string = data[::-1]

# Stop the timer
end = time.time()
duration = end - start

# Display the result and the time taken
print("Reversed string:", reversed_string)
print("Time taken:", duration, "seconds")

File: cpp/04.cpp
#include <iostream>
#include <string>
#include <chrono>

int main() {
    // Create a large string as sample data
    std::string data = "Lorem ipsum dolor sit amet, consectetur adipiscing elit.";

    // Start the timer
    auto start = std::chrono::high_resolution_clock::now();

    // Perform string processing operations here
    std::string reversed;
    for (int i = data.size() - 1; i >= 0; --i) {
        reversed += data[i];
    }

    // Stop the timer
    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> duration = end - start;

    // Display the result and the time taken
    std::cout << "Reversed string: " << reversed << std::endl;
    std::cout << "Time taken: " << duration.count() << " seconds" << std::endl;

    return 0;
}
